src/main/java/ca/corefacility/bioinformatics/irida/ria/web/analysis/AnalysisController.java

```diff
@@ -229,130 +230,215 @@ public String getUserAnalysisOutputsPage(Model model) {
 		}
 
-		return viewName;
+		inputFilesMap.put("sequenceFilePairList", seqFilePairs);
+		inputFilesMap.put("sequenceFileSizeList", seqFileSizes);
+
+		return inputFilesMap;
 	}
 
 	/**
-	 * Update an analysis name
+	 * Update an analysis name and/or priority
 	 *
-	 * @param submissionId ID of the submission to update
-	 * @param name         name to update the analysis to
-	 * @param priority     the priority to update the analysis to.  Note only admins will be allowed to update priority
-	 * @param model        model for view
-	 * @param locale       locale of the user
+	 * @param parameters parameters which include the submission id and the new name and/or priority
+	 * @param locale     User's locale
 	 * @return redirect to the analysis page after update
 	 */
-	@RequestMapping(value = "/{submissionId}/edit", produces = MediaType.TEXT_HTML_VALUE)
-	public String editAnalysis(@PathVariable Long submissionId, @RequestParam String name,
-			@RequestParam(required = false, defaultValue = "") AnalysisSubmission.Priority priority, Model model,
-			Locale locale) {
-		AnalysisSubmission submission = analysisSubmissionService.read(submissionId);
-
-		submission.setName(name);
-
-		boolean error = false;
-
-		try {
-			analysisSubmissionService.update(submission);
-			// Setting the priority as a separate call as it's not allowed to be updated with the normal update
-			if (priority != null) {
-				analysisSubmissionService.updatePriority(submission, priority);
-			}
-		} catch (Exception e) {
-			logger.error("Error while updating analysis", e);
-			error = true;
-		}
-
-		if (error) {
-			model.addAttribute("updateError", true);
-			return getDetailsPage(submissionId, model, locale);
+	@RequestMapping(value = "/ajax/update-analysis", method = RequestMethod.PATCH)
+	public Map<String, String> ajaxUpdateSubmission(@RequestBody AnalysisSubmissionInfo parameters, Locale locale) {
+		String message = "";
+		logger.trace("reading analysis submission " + parameters.getAnalysisSubmissionId());
+		AnalysisSubmission submission = analysisSubmissionService.read(parameters.getAnalysisSubmissionId());
+
+		if (parameters.getAnalysisName() != null) {
+			analysisSubmissionService.updateAnalysisName(submission, parameters.getAnalysisName());
+			logger.trace("Name updated for: " + submission);
+			message = messageSource.getMessage("analysis.notification.analysis.name.updated",
+					new Object[] { parameters.getAnalysisName() }, locale);
+		} else if (parameters.getPriority() != null) {
+			analysisSubmissionService.updatePriority(submission, parameters.getPriority());
+			logger.trace("Priority updated for: " + submission);
+			message = messageSource.getMessage("analysis.notification.analysis.priority.updated",
+					new Object[] { parameters.getPriority(), submission.getName() }, locale);
 		}
 
-		return "redirect:/analysis/" + submissionId;
+		return ImmutableMap.of("result", "success", "message", message);
```

Comment:
Same thing here with using `HttpServletResponse`

Reply:
Fixed in 9c4e640d5c452bf1455f9b93a1b1499a418283e6